sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/dao/room/ScheduleRepository.kt

```diff
@@ -155,33 +157,63 @@ open class ScheduleRepository constructor(
             return now().plusDays(cachedDaysAhead)
         }
 
+    /**
+     * @property asyncScheduler for performing network and database operations
+     */
+    @VisibleForTesting
+    protected open val asyncScheduler: Scheduler get() = Schedulers.io()
+
+    /**
+     * @property scheduleRepoErrorLiveData for monitoring error messages from the schedule repository
+     */
+    val scheduleRepoErrorLiveData = MutableLiveData<String>()
```

Comment:
@liujoshua I moved the error live data to the schedule repo so that the repository could be the one managing the subscriptions while still be able to communicate back to the view layer.